parser_library/src/semantics/operand_impls.cpp

```diff
@@ -90,17 +90,15 @@ std::unique_ptr<checking::operand> make_check_operand(expressions::mach_evaluate
     {
         return std::make_unique<checking::one_operand>(res.get_abs());
     }
+    else if (res.value_kind() == context::symbol_value_kind::RELOC && type_hint
+        && *type_hint == checking::machine_operand_type::RELOC_IMM)
+    {
+        return std::make_unique<checking::one_operand>("RELOC", res.get_reloc().offset());
```

Comment:
This code would accept instructions like:
```
RELOC LR 1,1
 EXRL 1,RELOC+5
 EXRL 1,RELOC+RELOC
```
The second one is definitely invalid, I am not sure about the first one.
We should generate this special operand only when the relocatable symbol is not complex (`res.get_reloc().is_simple()`) and maybe even just when it is just a simple symbol (not sure about the latter).

Reply:
the `exrl` takes the difference in halfwords (multiples of 2 bytes), so `reloc+5` cannot be represented at all.

Reply:
OK, but `RELOC+4` would be fine?

Reply:
yes, `reloc+4` is ok.

Reply:
Length being multiple of 2 bytes holds true only for exrl instruction or for all instructions with relative addressing?

Reply:
as far as I know, it's for all reladdr operands.